### ISSUE TITLE
Telnet Library: Allowing control over connection timeout

### ISSUE THREAD

**Issue by vkruoso** [robotframework/robotframework#2079]
When using the `open_connection` method of the Telnet library we can provide a `timeout` parameter. That value is used to control the timeout when executing commands (basically read functions). That timeout is only applied after the connection has been established:

```python
    telnetlib.Telnet.__init__(self, host, int(port) if port else 23)
    self._set_timeout(timeout)
```

As you can see, there is no way to provide a timeout to the connection establishment in the ``__init__`` call. Looks like the telnetlib will use the operating system global timeout ``socket._GLOBAL_DEFAULT_TIMEOUT`` that seems to be 2 minutes in my system.

That support should be easy to implement and I can provide a PR if you see that this can be merged.

**Comment by pekkaklarck** [comment]
Sounds good to me. Any objections by @jussimalinen? 

Remote library had same issue. You can take a look at how it was fixed.

**Comment by jussimalinen** [comment]
Sounds good.

**Comment by MoonSungHoon** [comment]
Sounds good. :+1:

**Comment by vkruoso** [comment]
Does it sounds reasonable to use the same `timeout` parameter as the connection timeout too? Creating a new `connection_timeout` parameter does not sound good, and also may cause confusion about the purpose of the `timeout` parameter itself. What to you think about it?

**Comment by pekkaklarck** [comment]
I'm worried that the current `timeout` is a bit short, only 3 seconds, to be used for connection. This is especially true if it would be used also when waiting replies in general, not only when connecting, similarly as Remote does. This change would also be backwards incompatible without an easy to keep using system's default timeout.

**Comment by vkruoso** [comment]
That is true and if that was the approach chosen I would suggest increasing the default timeout. Anyway, how does a `connection_timeout` parameter sounds? Can you think on a better name? If you can't, I can go ahead on implementing, improve docs and add tests.

**Comment by pekkaklarck** [comment]
Increasing the default timeout would be backwards incompatible and I'm afraid we needed to increase it quite a lot.

`connection_timeout` sounds good. I doubt there's a need for a separate keyword for setting it, but I think the library itself should accept the default value and `Open Connection` should allow overriding it.

**Comment by jussimalinen** [comment]
Implemented in #2119
